### ISSUE TITLE
The spec doesn't describe the newest cast semantics

### ISSUE THREAD

**Issue by dimvar** [microsoft/TypeScript#12419]
When casting from a type A to a type B, the spec says that either A is assignable to B, or B is assignable to A. But the playground allows something more general: in some cases, as long as A and B have a common subtype, it doesn’t warn. The next example is valid.

````ts
function f(x: number | string) {
    return <number | boolean>(x);
}
````

The playground uses the [comparable relation](https://github.com/Microsoft/TypeScript/pull/5517). The spec should be updated to include it.

**Comment by RyanCavanaugh** [comment]
The spec is now an "archived" artifact and we won't be tracking defects against it anymore.
